smartsim/_core/_cli/utils.py

```diff
@@ -81,3 +85,100 @@ def pip_install(packages: t.List[str], end_point: t.Optional[str] = None, verbos
         raise BuildError(error)
     if verbose:
         logger.info(f"{packages} installed successfully")
+
+
+def clean(core_path: Path, _all: bool = False) -> int:
+    """Remove pre existing installations of ML runtimes
+
+    :param _all: Remove all non-python dependencies
+    :type _all: bool, optional
+    """
+
+    build_temp = core_path / ".third-party"
+    if build_temp.is_dir():
+        shutil.rmtree(build_temp, ignore_errors=True)
+
+    lib_path = core_path / "lib"
+    if lib_path.is_dir():
+
+        # remove RedisAI
+        rai_path = lib_path / "redisai.so"
+        if rai_path.is_file():
+            rai_path.unlink()
+            logger.info("Successfully removed existing RedisAI installation")
+
+        backend_path = lib_path / "backends"
+        if backend_path.is_dir():
+            shutil.rmtree(backend_path, ignore_errors=True)
+            logger.info("Successfully removed ML runtimes")
+
+    bin_path = core_path / "bin"
+    if bin_path.is_dir() and _all:
+        files_to_remove = ["redis-server", "redis-cli", "keydb-server", "keydb-cli"]
+        removed = False
+        for _file in files_to_remove:
+            file_path = bin_path.joinpath(_file)
+
+            if file_path.is_file():
+                removed = True
+                file_path.unlink()
+        if removed:
+            logger.info("Successfully removed SmartSim database installation")
+
+    return 0
+
+
+def get_db_path() -> t.Optional[Path]:
+    bin_path = get_install_path() / "_core" / "bin"
+    for option in bin_path.iterdir():
+        if option.name in ("redis-cli", "keydb-cli"):
+            return option
+    return None
+
+
+CliHandler = t.Callable[[Namespace], int]
+CliParseConfigurator = t.Callable[[ArgumentParser], None]
```

Comment:
Incredibly pedantic request: If these types are not meant to be used as annotations outside of this file can we `_` prefix them (not that it matters to an end user bc this is all in a "protected " `_core` dir anyway)

Reply:
Done! I was originally thinking that I might import these into the implementations of each menu item and that evaporated.